fastlane_bot/tools/cpc.py

```diff
@@ -1067,15 +1075,21 @@ def from_carbon(
                 if not tkny == tknq:
                     pa, pb = 1 / pa, 1 / pb
 
-            # zero-width ranges are somewhat extended for numerical stability
+            # small and zero-width ranges are extended for numerical stability
             pa0, pb0 = pa, pb
+            if pa/pb-1 < minrw:
+                pa = pb = sqrt(pa*pb)
+                assert pa == pb, "just making sure"
             if pa == pb:
-                pa *= 1.0000001
-                pb /= 1.0000001
+                # pa *= 1.0000001
+                # pb /= 1.0000001
+                rw_multiplier = sqrt(1+minrw)
+                pa *= rw_multiplier
+                pb /= rw_multiplier
 
             # validation
-            if not pa > pb:
-                raise cls.CPCValidationError(f"pa > pb required ({pa}, {pb})")
+            if not pa/pb - 1 >= minrw*0.99:
```

Comment:
I'm pretty sure that you should use `abs(pa/pb - 1)` here.

Reply:
no; as you see I am testing here for pa > pb; I am just testing with a small tolerance, eg if the ranges have been created in a loop